specification/hugr.md

```diff
@@ -1580,6 +1580,8 @@ struct HUGR {
 struct Node{
   // parent node index
   parent: Int,
+  // The input resources to the node
+  input_resources: Option<ResourceSet>
```

Comment:
I thinki this is a bit neater in terms of the serialised format than embedding a Nodetype. I assume the serde serialisation is being deferred to the dispatchees of `OpType` and can vary in length a fair bit, so it's nicer to have it at the end imo